File: backend/grabtable/src/main/java/edu/skku/grabtable/reservation/domain/response/ReservationDetailResponse.java
package edu.skku.grabtable.reservation.domain.response;

import edu.skku.grabtable.order.domain.response.OrderResponse;
import edu.skku.grabtable.order.domain.response.SharedOrderResponse;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
@Getter
public class ReservationDetailResponse {

    private Long id;
    private Long storeId;
    private UserCartsInfoResponse host;
    private List<UserCartsInfoResponse> invitees;
    private String inviteCode;
    private SharedOrderResponse sharedOrder;
    private List<OrderResponse> orders;

    public static ReservationDetailResponse of(
            Long reservationId,
            Long storeId,
            UserCartsInfoResponse hostInfo,
            List<UserCartsInfoResponse> inviteesInfo,
            String inviteCode,
            List<OrderResponse> orders
    ) {
        return new ReservationDetailResponse(
                //TODO
                reservationId,
                storeId,
                hostInfo,
                inviteesInfo,
                inviteCode,
                null,
                orders
        );
    }
}

File: frontend/grabtable/src/app/types/reservationDetailResponse.ts
import { OrderResponse } from './orderResponse'
import { SharedOrderResponse } from './sharedOrderResponse'
import { UserCartsInfoResponse } from './userCartsInfoResponse'

export type ReservationDetailResponse = {
  id: number
  storeId: number
  host: UserCartsInfoResponse
  invitees: UserCartsInfoResponse[]
  inviteCode: string
  sharedOrder: SharedOrderResponse
  orders: OrderResponse[]
}
